webapp/src/Controller/BaseController.php

```diff
@@ -369,24 +382,28 @@ protected function deleteEntities(
 
         [$isError, $primaryKeyData, $deleteTreeMessages] = $this->buildDeleteTree($entities, $relations, $entityManager);
         if (!empty($deleteTreeMessages)) {
-            $messages = $deleteTreeMessages;
+            $messages = array_unique($deleteTreeMessages);
         }
 
-        if ($request->isMethod('POST')) {
+        // Check if we handle a list of deletable entities.
+        $removeList = $request->request->get('confirmation', true);
+
+        if ($request->isMethod('POST') && $removeList) {
             if ($isError) {
                 throw new BadRequestHttpException(reset($messages));
             }
 
-            $msgList = [];
             foreach ($entities as $id => $entity) {
+                $msgList = [];
```

Comment:
So basically we do the eventlog after this loop with array_keys($id) as argument. That will also speed up deletion